dietpi/dietpi-software

```diff
@@ -533,22 +540,135 @@ INDEX_WEBSERVER_TARGET=${INDEX_WEBSERVER_TARGET:=-2}
+                                # Default to LXDE
+                                DESKTOP_LXDE=1
+                        elif (( $DESKTOP_MATE == 1 )) && (( $DESKTOP_GNUSTEP == 0 )); then
+                                DESKTOP_LXDE=0
+                                DESKTOP_MATE=1
+                                DESKTOP_GNUSTEP=0
+                        elif (( $DESKTOP_MATE == 0 )) && (( $DESKTOP_GNUSTEP == 1 )); then
+                                DESKTOP_LXDE=0
+                                DESKTOP_MATE=0
+                                DESKTOP_GNUSTEP=1
+                        fi
 		fi
 
+                #Installs that require a DESKTOP
+                if (( $QUITERSS == 1 )); then
+                        if (( $DESKTOP_LXDE == 0 )) && (( $DESKTOP_MATE == 0 )) && (( $DESKTOP_GNUSTEP == 0 )); then
+                                # Default to LXDE
+                                DESKTOP_LXDE=1
+                        elif (( $DESKTOP_MATE == 1 )) && (( $DESKTOP_GNUSTEP == 0 )); then
+                                DESKTOP_LXDE=0
+                                DESKTOP_MATE=1
+                                DESKTOP_GNUSTEP=0
+                        elif (( $DESKTOP_MATE == 0 )) && (( $DESKTOP_GNUSTEP == 1 )); then
+                                DESKTOP_LXDE=0
+                                DESKTOP_MATE=0
+                                DESKTOP_GNUSTEP=1
+                        fi
+                fi
+
 		#Desktop LXDE
 		if (( $DESKTOP_LXDE == 1 )); then
-
 			INSTALL_DESCRIPTION='Desktop (LXDE)'
 			Banner_Installing
 
 			AGI lxde upower policykit-1 iceweasel p7zip-full --no-install-recommends
 			#upower policykit-1. Needed for LXDE logout menu item to show shutdown/restart ......
+		fi
+
+		#Desktop MATE
+		if (( $DESKTOP_MATE == 1 )); then
+			INSTALL_DESCRIPTION="Desktop (MATE)"
+			Banner_Installing
+
+			AGI mate-desktop-environment-extras upower policykit-1 iceweasel p7zip-full --no-install-recommends
+		fi
+
+		#Desktop GNUStep
+                if (( $DESKTOP_GNUSTEP == 1 )); then
+                        INSTALL_DESCRIPTION="Desktop (GNUStep)"
+                        Banner_Installing
+
+                        AGI x-window-system-core wmaker gnustep gnustep-devel gnustep-games libc-dbg upower policykit-1 iceweasel p7zip-full --no-install-recommends
+                fi
+
+		#QuiteRSS
+                if (( $QUITERSS == 1 )); then
+                        INSTALL_DESCRIPTION="QuiteRSS"
+                        Banner_Installing
+
+                        AGI quiterss
+                fi
+
+		#XRDP
+		if (( $XRDP == 1 )); then
+                        INSTALL_DESCRIPTION="XRDP (Remote Desktop Server)"
+                        Banner_Installing
 
+                        AGI xrdp
+                fi
+
+		#NOMACHINE
+		if (( $NOMACHINE == 1 )); then
+			INSTALL_DESCRIPTION="NoMachine (Remote Desktop Server/Client)"
+			Banner_Installing
+
+			## HW MODELS NEED TO BE CONSIDERED
```

Comment:
Yep :+1: Probably best to use the new ```HW_ARCH``` identifier: https://github.com/Fourdee/DietPi/blob/testing/dietpi/dietpi-obtain_hw_model#L39-L46

Example usage: https://github.com/Fourdee/DietPi/blob/testing/dietpi/dietpi-software#L1999-L2018